lib/Test/Consul.pm

```diff
@@ -13,17 +12,130 @@ use Carp qw(croak);
 use HTTP::Tiny;
 use Net::EmptyPort qw( check_port );
 use File::Temp qw( tempfile );
+use Scalar::Util qw( blessed );
+
+use Moo;
+use Types::Standard qw( Bool Enum Undef );
+use Types::Common::Numeric qw( PositiveInt );
+use Types::Common::String qw( NonEmptySimpleStr );
+
+my $start_port = 49152;
+my $current_port = $start_port;
+my $end_port  = 65535;
+
+sub _unique_empty_port {
+    my ($udp_too) = @_;
+
+    my $port = 0;
+    while ($port == 0) {
+        $current_port ++;
+        $current_port = $start_port if $current_port > $end_port;
+        next if check_port( undef, $current_port, 'tcp' );
+        next if $udp_too and check_port( undef, $current_port, 'udp' );
+        $port = $current_port;
+    }
+
+    # Make sure we return a scalar with just numeric data so it gets
```

Comment:
The idiomatic way to forcibly numerify a value is `0+$port`. Not needed here, just responding to the comment.